subsys/net/lib/sockets/Kconfig

```diff
@@ -48,6 +48,7 @@ config NET_SOCKETS_DNS_TIMEOUT
 
 config NET_SOCKETS_SOCKOPT_TLS
 	bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
+	depends on MBEDTLS
```

Comment:
Actually I'd make it: `select MBEDTLS if NET_NATIVE`

I saw that `NET_SOCKETS_SOCKOPT_TLS` option is also used in some offloaded engines (SimpleLink for instance) as well as in some application protocols (like LWM2M). mbedTLS is only needed for native implementation.